apps/src/aiDifferentiation/AiDiffContainer.tsx

```diff
@@ -53,10 +53,22 @@ const AiDiffContainer: React.FC<AiDiffContainerProps> = ({
     state => state.currentUser.hasCompletedAiDifferentiationWelcome
   );
 
+  const MIN_VISIBLE = 40;
+
   useEffect(() => {
     const ensureDraggableIsVisible = () => {
-      if (window.innerWidth < positionX + 100) {
-        setPositionX(window.innerWidth - 100);
+      const boxWidth = parseInt(style.containerWidth);
+      const originX = parseInt(style.fabOriginX);
+      // These isNaN checks are for testing as the SCSS variables don't come
+      // through properly in the test environment.
+      const minX = isNaN(originX) ? 0 : MIN_VISIBLE - originX - boxWidth;
+      const maxX = isNaN(originX)
+        ? 1000
+        : document.documentElement.clientWidth - originX - MIN_VISIBLE;
```

Comment:
We can move these definitions to constants so that they don't update every time positionX changes.